src/waypoint.jsx

```diff
@@ -333,13 +352,20 @@ export default class Waypoint extends React.Component {
    * @return {Object}
    */
   render() {
-    // We need an element that we can locate in the DOM to determine where it is
-    // rendered relative to the top of its context.
-    return <span ref={this.refElement} style={{ fontSize: 0 }} />;
+    if (this.props.children) {
+      // TODO: consider returning children directly when there is only one child,
+      // using React.cloneElement to attach a ref (but child might already have a ref)
```

Comment:
😄 awesome!

@jmeas I don't think that's allowed in any case – https://github.com/facebook/react/issues/2127. IIRC, `children` is either an array or a single element (not wrapped in an array), so when it's a single element, you can return it directly (after adding a `ref` attribute). But that's a lot of work for a potentially limited usecase